### ISSUE TITLE
DLA Engine Build Failed on Orin

### ISSUE THREAD

**Issue by Nik-V9** [NVIDIA/TensorRT#2426]
## Description

Building a DLA Engine using Torch_TensorRT failed on the Orin with the following result:

```
ERROR: [Torch-TensorRT TorchScript Conversion Context] - 1: [nvdlaUtils.cpp::submit::198] Error Code 1: DLA (Failure to submit program to DLA engine.)
ERROR: [Torch-TensorRT TorchScript Conversion Context] - 2: [builder.cpp::buildSerializedNetwork::620] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
RuntimeError: [Error thrown at core/conversion/conversionctx/ConversionCtx.cpp:147] Building serialized network failed in TensorRT
```
Trying to build HRNet-W32 model on DLA with a input shape of [1, 2, 2048, 2560].

## Environment

**TensorRT Version**: 8.4.0-1+cuda11.4
**JetPack Version**: 5.0.1dp
**NVIDIA GPU**:  Orin
**NVIDIA Driver Version**: 
**CUDA Version**:  11.4
**CUDNN Version**: 
**Operating System**:  Ubuntu 20.04.5 LTS

## Relevant Files

The Code is available in this repo: https://github.com/castacks/daa_profile

Compiling the TRT Engine on DLA here: https://github.com/castacks/daa_profile/blob/b7c592788671101821cad9a01fc27539f72e22e5/model.py#L103

## Steps To Reproduce

- Clone Github Repo on Orin. 
- Follow Download Instructions in Github ReadME and unzip files on Orin.
- Run the following command within the repo: `python3 profile.py profile.yaml`

```
Running exp profile.yaml
Loaded config:
{'input_dir': '/home/airlab/DAA/extrinsic', 'num_cam': 1, 'trt_batch_size': 1, 'width': 2448, 'height': 2048, 'padding': 56, 'use_tensorrt': True, 'use_torch2trt': False, 'use_dla': True, 'input_frames': 2, 'models_dir': '/home/airlab/DAA/models', 'full_res_model_chkpt': '120_hrnet32_all_2220.pth', 'fp16_mode': True, 'int8_mode': False}
Compiling TensorRT detector model ...
WARNING: [Torch-TensorRT] - Setting GPU id to 0 for device because device 0 manages DLA on Xavier
WARNING: [Torch-TensorRT TorchScript Conversion Context] - DLA LAYER: CBUF size requirement for layer %input.87 : Tensor = aten::_convolution(%5626, %self.base_model.transition1.0.0.weight, %4054, %7, %7, %7, %10, %6, %11, %9, %10, %9, %9), scope: __module.base_model/__module.base_model.transition1.0/__module.base_model.transition1.0.0 # /home/airlab/.local/lib/python3.8/site-packages/torch/nn/modules/conv.py:443:0 exceeds the limit.
WARNING: [Torch-TensorRT TorchScript Conversion Context] - Default DLA is enabled but layer %input.87 : Tensor = aten::_convolution(%5626, %self.base_model.transition1.0.0.weight, %4054, %7, %7, %7, %10, %6, %11, %9, %10, %9, %9), scope: __module.base_model/__module.base_model.transition1.0/__module.base_model.transition1.0.0 # /home/airlab/.local/lib/python3.8/site-packages/torch/nn/modules/conv.py:443:0 is not supported on DLA, falling back to GPU.
WARNING: [Torch-TensorRT TorchScript Conversion Context] - DLA LAYER: CBUF size requirement for layer %input.91 : Tensor = aten::_convolution(%5626, %self.base_model.transition1.1.0.0.weight, %4054, %4055, %7, %7, %10, %6, %11, %9, %10, %9, %9), scope: __module.base_model/__module.base_model.transition1.1/__module.base_model.transition1.1.0/__module.base_model.transition1.1.0.0 # /home/airlab/.local/lib/python3.8/site-packages/torch/nn/modules/conv.py:443:0 exceeds the limit.
WARNING: [Torch-TensorRT TorchScript Conversion Context] - Default DLA is enabled but layer %input.91 : Tensor = aten::_convolution(%5626, %self.base_model.transition1.1.0.0.weight, %4054, %4055, %7, %7, %10, %6, %11, %9, %10, %9, %9), scope: __module.base_model/__module.base_model.transition1.1/__module.base_model.transition1.1.0/__module.base_model.transition1.1.0.0 # /home/airlab/.local/lib/python3.8/site-packages/torch/nn/modules/conv.py:443:0 is not supported on DLA, falling back to GPU.
ERROR: [Torch-TensorRT TorchScript Conversion Context] - 1: [nvdlaUtils.cpp::submit::198] Error Code 1: DLA (Failure to submit program to DLA engine.)
ERROR: [Torch-TensorRT TorchScript Conversion Context] - 2: [builder.cpp::buildSerializedNetwork::620] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
Traceback (most recent call last):
  File "profile.py", line 127, in <module>
    run_evaluation(args.experiment)
  File "profile.py", line 103, in run_evaluation
    detector = SegDetector(cfg=dict(
  File "/home/airlab/DAA/daa_profile/model.py", line 103, in __init__
    trt_engine = torch_tensorrt.ts.convert_method_to_trt_engine(
  File "/home/airlab/.local/lib/python3.8/site-packages/torch_tensorrt/ts/_compiler.py", line 199, in convert_method_to_trt_engine
    return _C.convert_graph_to_trt_engine(module._c, method_name, _parse_compile_spec(compile_spec))
RuntimeError: [Error thrown at core/conversion/conversionctx/ConversionCtx.cpp:147] Building serialized network failed in TensorRT
```

**Comment by zerollzeng** [comment]
Usually it's cause by some DLA bad layers, I'm not an expert in Torch-TRT, could you please export the model to ONNX and try it with trtexec(I think it should be like `/usr/src/tensorrt/bin/trtexec --onnx=model.onnx --fp16 --useDLACore=0 --allowGPUFallback`)? thanks!

**Comment by Nik-V9** [comment]
Hi, Thanks for the follow up!

I converted the pytorch model to ONNX and tried building the engine with trtexec. I got the same error as Torch-TensorRT.

`trtexec --onnx=model_b_1.onnx --fp16 --useDLACore=0 --allowGPUFallback --saveEngine=model_onnx_dla_b_1.engine`

The ONNX model is available here:

**Comment by zerollzeng** [comment]
```
[11/02/2022-11:46:03] [I] === Performance summary ===
[11/02/2022-11:46:03] [I] Throughput: 0.406047 qps
[11/02/2022-11:46:03] [I] Latency: min = 2240.97 ms, max = 2241.41 ms, mean = 2241.14 ms, median = 2241.13 ms, percentile(90%) = 2241.2 ms, percentile(95%) = 2241.41 ms, percentile(99%) = 2241.41 ms
[11/02/2022-11:46:03] [I] Enqueue Time: min = 0.967136 ms, max = 1.2666 ms, mean = 1.14105 ms, median = 1.13232 ms, percentile(90%) = 1.24414 ms, percentile(95%) = 1.2666 ms, percentile(99%) = 1.2666 ms
[11/02/2022-11:46:03] [I] H2D Latency: min = 1.78027 ms, max = 1.9176 ms, mean = 1.80922 ms, median = 1.80225 ms, percentile(90%) = 1.80664 ms, percentile(95%) = 1.9176 ms, percentile(99%) = 1.9176 ms
[11/02/2022-11:46:03] [I] GPU Compute Time: min = 2239.08 ms, max = 2239.32 ms, mean = 2239.16 ms, median = 2239.15 ms, percentile(90%) = 2239.22 ms, percentile(95%) = 2239.32 ms, percentile(99%) = 2239.32 ms
[11/02/2022-11:46:03] [I] D2H Latency: min = 0.101562 ms, max = 0.177734 ms, mean = 0.16792 ms, median = 0.174805 ms, percentile(90%) = 0.177734 ms, percentile(95%) = 0.177734 ms, percentile(99%) = 0.177734 ms
[11/02/2022-11:46:03] [I] Total Host Walltime: 24.6277 s
[11/02/2022-11:46:03] [I] Total GPU Compute Time: 22.3916 s
[11/02/2022-11:46:03] [I] Explanations of the performance metrics are printed in the verbose logs.
[11/02/2022-11:46:03] [I] 
&&&& PASSED TensorRT.trtexec [TensorRT v8501] # trtexec --onnx=model_b_1.onnx --fp16 --useDLACore=0 --allowGPUFallback
```
In my test the issue is fixed in the latest TRT 8.5. can you wait for the new Jetpack release that contains it? thanks!

**Comment by zerollzeng** [comment]
The next JP 5.1 I think, but I'm not sure about the exact release date. please stay in tune :-)
